# Statistical Inference Exercise Solutions

# Introduction

**Question** - The goal of statistical inference is to? 
1. Infer facts about a population from a sample.
2. Infer facts about the sample from a population.
3. Calculate sample quantities to understand your data.
4. To torture Data Science students.

1

**Question** - The goal of randomization of a treatment in a randomized trial is to?
1. It doesn’t really do anything.
2. To obtain a representative sample of subjects from the population of interest.
3. Balance unobserved covariates that may contaminate the comparison between the
    treated and control groups.
4. To add variation to our conclusions.


3

**Question** - Probability is a?
1. Population quantity that we can potentially estimate from data.
2. A data quantity that does not require the idea of a population.

1

# Probability

**Question** - Can you add the probabilities of any two events to get the probability of at least one occurring?

Yes if they are mutually exclusive.

**Question** -  I define a PMF, p so that for x = 0 and x = 1 we have p(0) = -0.1 and p(1) = 1.1. Is this a
valid PMF?

No.  For a valid PMF p(x) must be greater than or equal to 0 for all x.

$p(x) \ge 0$ for all x.

**Question** -  What is the probability that 75% or fewer calls get answered in a randomly sampled day from
the population distribution from this chapter?

$p(x < 0.75) = \int_0^{0.75}2x dx = 0.75^2 = 0.5625 $

Or exploiting the fact that it is a triangle:

In [7]:
1.5 * 0.75 * 0.5

0.5625

**Question** -  The 97.5th percentile of a distribution is?

The probability of a random sample being less than or equal to 0.975.

$p(x \le 0.975)$

**Question** -  Consider influenza epidemics for two parent heterosexual families. Suppose that the probability
is 15% that at least one of the parents has contracted the disease. The probability that the
father has contracted influenza is 10% while that the mother contracted the disease is 9%. What
is the probability that both contracted influenza expressed as a whole number percentage?

$P(A\cup B) = P(A) + P(B) - P(A\cap B)$ thus $$0.15 = 0.10 + 0.09 - P(A\cap B)$$

0.10 + 0.09 - 0.15 = 0.04

**Question** -  A random variable, X, is uniform, a box from 0 to 1 of height 1. (So that it’s density is f(x) = 1
for 0 <= x <= 1.) What is it’s median expressed to two decimal places?

The median is the same as the 50th percentile so,

$p(X \le 0.5) = \int_0^{a} dx$ = a - 0 = 0.5

This density looks like a box. So, notice that $P(X \leq x) = width\times height = x$. We want $.5 = P(X\leq x) = x$.

**Question** -  If a continuous density that never touches the horizontal axis is symmetric about zero, can
we say that its associated median is zero?

Yes.

This is a surprisingly hard problem. The easy explanation is that 50% of the probability is below 0 and 50% is above so yes. However, it is predicated on the density not being a flat line at 0 around 0. That's why the caveat that it never touches the horizontal axis is important.

$p(x) = p(-x)$ for all x, and since $\int_{-\infty}^{\infty} p(x)dx = 1$ then it follows that $\int_{-\infty}^{0}p(x)dx = \int_0^{\infty}p(x)dx = \frac{1}{2} $

# Conditional Probability

**Question** - The odds associated with a probability, p, are defined as? I pull a card from a deck and do not show you the result. I say that the resulting card is a
heart. What is the probability that it is the queen of hearts?

1/13 since it is already known to be a heart (unless I am a liar of course).

**Question** - The odds associated with a probability, p, are defined as?

$Odds = \frac{p}{1 - p}$

**Question** - The probability of getting two sixes when rolling a pair of dice is?

Let A be prob of dice A landing 6 and B be prob of dice B landing 6.  Since A and B are independent, then

$P(A \cup B) = P(A)P(B) = \frac{1}{6}\frac{1}{6} = 0.028$

**Question** - The probability that a manuscript gets accepted to a journal is 12% (say). However, given
that a revision is asked for, the probability that it gets accepted is 90%. Is it possible that the
probability that a manuscript has a revision asked for is 20%?

P(A∩B)=P(A|B)∗P(B)=.9×.2=.18 this is larger than P(A)=.12, which is not possible since A∩B⊂A

**Question** - Suppose 5% of housing projects have issues with asbestos. The sensitivity of a test for asbestos
is 93% and the specificity is 88%. What is the probability that a housing project has no asbestos
given a negative test expressed as a percentage to the nearest percentage point?

$P(A^c|T-)=\frac{P(T−|A^c)P(A^c)}{P(T−|A^c)P(A^c)+P(T−|A)P(A)}$

In [13]:
(.88 * .95) / (.88 * .95 + .07 * .05)

0.995830851697439

# Expected Values

**Question** - A standard die takes the values 1, 2, 3, 4, 5, 6 with equal probability. What is the expected
value?

$E[X] = \sum_{i=1}^{n} x_i p(x_i) = \frac{1}{6}(1 + 2 + 3 + 4 + 5 + 6) = \frac{21}{6} = 3.5$

**Question** - Consider a density that is uniform from -1 to 1. (I.e. has height equal to 1/2 and looks like a
box starting at -1 and ending at 1). What is the mean of this distribution?

Since it is symetric about the vertical axis, the mean is 0.

**Question** - If a population has mean u, what is the mean of the distribution of averages of 20 observations
from this distribution?

u

**Question** - You are playing a game with a friend where you flip a coin and if it comes up heads you give
her X dollars and if it comes up tails she gives you $Y$ dollars. The odds that the coin is heads
is d. What is your expected earnings?

You lose X with probability $p = \frac{d}{(1+d)}$ and you win Y with probability $1 − p = \frac{1}{(1+d)}$. So your answer is
$−X\frac{d}{1+d}+Y\frac{1}{1+d}$

**Question** - If you roll ten standard dice, take their average, then repeat this process over and over and
construct a histogram what would it be centered at?

3.5

# Variation

**Question** - If I have a random sample from a population, the sample variance is an estimate of?
1. The population standard deviation.
2. The population variance.
3. The sample variance.
4. The sample standard deviation.

2

**Question** - The distribution of the sample variance of a random sample from a population is centered at
what?
1. The population variance.
2. The population mean.

1

**Question** - I keep drawing samples of size n from a population with variance 2 and taking their average.
I do this thousands of times. If I were to take the variance of the collection of averages, about
what would it be?

$\frac{\sigma^2}{n} = \frac{2}{n} $ Where n is the number of samples

**Question** - You get a random sample of **n** observations from a population and take their average. You
would like to estimate the variability of averages of **n** observations from this population
to better understand how precise of an estimate it is. Do you need to repeated collect averages
to do this?

1. No, we can multiply our estimate of the population variance by 1/n to get a good estimate of the variability of the average.
2. Yes, you have to get repeat averages.

1

**Question** - A random variable takes the value -4 with probability .2 and 1 with probability .8. What is
the variance of this random variable?

$E[X]=0 \\
Var(X)=E[X2]=(−4)2∗.2+(1)2∗.8 = 4$

In [3]:
-4 * .2 + 1 * .8

0.0

In [5]:
(-4)**2 * .2 + (1)**2 * .8

4.0

**Question** - Let X be a random variable having standard deviation . What can be said about the variance
of X/o?

It has variance 1.

$ Var(\frac{X}{\sigma}) = \frac{Var(X)}{\sigma^2} = 1$

**Question** - If you roll ten standard dice, take their average, then repeat this process over and over and
construct a histogram, what would be its variance expressed to 3 decimal places?

The answer will be 0.292 since the variance of the sampling distribution of the mean is $\frac{\sigma^2}{10}$ where $\sigma^2$ is the variance of a single die roll, which is

In [23]:
import numpy as np
np.mean(((np.array(range(1, 7), dtype = 'float') - 3.5)**2)/10)

0.29166666666666669

# Common Distributions

**Question** -  Your friend claims that changing the font to comic sans will result in more ad revenue on your
web sites. When presented in random order, 9 pages out of 10 had more revenue when the
font was set to comic sans. If it was really a coin flip for these 10 sites, what’s the probability
of getting 9 or 10 out of 10 with more revenue for the new font?

$\binom{10}{9}(0.5)^9 (1 - 0.5)^{10 - 9} = 10(0.5)^{10} = 0.0098 $

In [25]:
10*0.5**10

0.009765625

**Question** - A software company is doing an analysis of documentation errors of their products. They sampled their very large codebase in chunks and found that the number of errors per chunk was approximately normally distributed with a mean of 11 errors and a standard deviation of 2. When randomly selecting a chunk from their codebase, whats the probability of fewer than 5 documentation errors?

In [41]:
from scipy.stats import norm
Z = (5 - 11) / 2
norm.cdf(Z)

#or alternatively
myNormal = norm(loc = 11, scale = 2)
myNormal.cdf(5)

0.0013498980316300933

**Question** - The number of search entries entered at a web site is Poisson at a rate of 9 searches per minute.
The site is monitored for 5 minutes. What is the probability of 40 or fewer searches in that
time frame?

In [48]:
from scipy.stats import poisson
myPoisson = poisson(mu = 9*5)
myPoisson.cdf(40)

0.25554507792812953

**Question** - Suppose that the number of web hits to a particular site are approximately normally
distributed with a mean of 100 hits per day and a standard deviation of 10 hits per day. What’s
the probability that a given day has fewer than 93 hits per day expressed as a percentage to
the nearest percentage point?

This can be estimated quickly to be approximately 0.26 since 10 hits is about one standard deviation from the mean so 50% - 34% = 26%

In [57]:
#or alternatively
myNormal = norm(loc = 100, scale = 10)
myNormal.cdf(93)

0.24196365222307303

**Question** - Suppose that the number of web hits to a particular site are approximately normally
distributed with a mean of 100 hits per day and a standard deviation of 10 hits per day. What
number of web hits per day represents the number so that only 5% of days have more hits?

In [61]:
myNormal = norm(loc = 100, scale = 10)
myNormal.ppf(0.95)

116.44853626951472

**Question** - Suppose that the number of web hits to a particular site are approximately normally
distributed with a mean of 100 hits per day and a standard deviation of 10 hits per day. Imagine
taking a random sample of 50 days. What number of web hits would be the point so that only
5% of averages of 50 days of web traffic have more hits?

In [63]:
myNormal = norm(loc = 100, scale = 10/(50)**0.5)
myNormal.ppf(0.95)

102.32617430735334

**Question** - You don’t believe that your friend can discern good wine from cheap. Assuming that you’re
right, in a blind test where you randomize 6 paired varieties (Merlot, Chianti, …) of cheap and
expensive wines. What is the change that she gets 5 or 6 right?

In [71]:
from scipy.stats import binom
myBinom = binom(n = 6, p = 0.5)
1 - myBinom.cdf(4)

0.109375

**Question** - The number of web hits to a site is Poisson with mean 16.5 per day. What is the probability of
getting 20 or fewer in 2 days?

In [72]:
myPoisson = poisson(mu = 16.5*2)
myPoisson.cdf(20)

0.010453906317688662

# Asymptopia

**Question** - I simulate 1,000,000 standard normals. The LLN says that their sample average must be close
to?

0

**Question** - About what is the probability of getting 45 or fewer heads out 100 flips of a fair coin? (Use the
CLT, not the exact binomial calculation).

In [7]:
import numpy as np
from scipy.stats import norm

myNorm = norm(loc = 0.50, scale = np.sqrt(0.5*0.5*100))
norm.cdf(0.45)

0.67364477971208003

**Question** - Consider the [Galton's Height Data](http://www.math.uah.edu/stat/data/Galton.html). Using the CLT and assuming that the fathers are a random sample from a population of interest, what is a 95% confidence mean height in inches?

In [13]:
import pandas as pd

Galton_data = pd.read_csv('../data/Galton.csv', index_col = None)
np.mean(Galton_data['Father']) + np.array([-1, 1]) * 1.645 * np.std(Galton_data['Father']/np.sqrt(len(Galton_data['Father'])))

array([ 69.09732319,  69.36837837])

**Question** - The goal of a a confidence interval having coverage 95% is to imply that:
1. If one were to repeated collect samples and reconstruct the intervals, around 95% percent
of them would contain the true mean being estimated.
2. The probability that the sample mean is in the interval is 95%.

In [ ]:
1

**Question** - The rate of search entries into a web site was 10 per minute when monitoring for an hour. Use
Python to calculate the exact Poisson interval for the rate of events per minute?

In [14]:
# Will assume a 95% CI
x = 10
t = 60
m = x/t
m + np.array([-1, 1]) * 1.645 * np.sqrt(m/t)

array([ 0.07996755,  0.25336578])

**Question** - Consider a uniform distribution. If we were to sample 100 draws from a a uniform distribution
(which has mean 0.5, and variance 1/12) and take their mean, $\bar{X}$. What is the approximate
probability of getting as large as 0.51 or larger?

In [19]:
from scipy.stats import norm
myNormal = norm(loc = 0.5, scale = np.sqrt(1 / 12 / 100))
1 - myNormal.cdf(0.51)

0.36451724476940184

# t Confidence Intervals

For iid Gaussian data, the statistic $\frac{\bar X - \mu}{s / \sqrt{n}}$ must follow a:
1. Z distribution
2. t distribution

2

**Question** - Paired differences T confidence intervals are useful when:
1. Pairs of observations are linked, such as when there is subject level matching or in a
study with baseline and follow up measurements on all participants.
2. When there was randomization of a treatment between two independent groups.



In [20]:
2

2

**Question** - The assumption that the variances are equal for the independent group T interval means that:
1. The sample variances have to be nearly identical.
2. The population variances are identical, but the sample variances may be different.

In [ ]:
1

**Question** - Load the data set mtcars in the from the ggplot module. Calculate a 95% confidence interval to
the nearest MPG for the variable mpg.

In [43]:
import statsmodels.stats.api as sm
import pandas as pd
from ggplot import mtcars as mtcars

sm.DescrStatsW(mtcars['mpg']).tconfint_mean()

(17.917678508746242, 22.263571491253757)

**Question** - Suppose that standard deviation of 9 paired differences is $1$. What value would the average
difference have to be so that the lower endpoint of a 95% students t confidence interval touches
zero?

We want $\bar{x} = t_{975,8}S/\sqrt{n}$

In [52]:
from scipy.stats import t

myT = t(df = 8)
myT.ppf(0.975) * 1/3

0.76866804501112351

**Question** - An independent group Student’s T interval is used instead of a paired T interval when:
1. The observations are paired between the groups.
2. The observations between the groups are naturally assumed to be statistically independent.
3. As long as you do it correctly, either is fine.
4. More details are needed to answer this question.

2
We can't pair them if the groups are independent of each other as well as independent within themselves.

**Question** - Consider the mtcars dataset. Construct a 95% T interval for MPG comparing 4 to 6 cylinder
cars (subtracting in the order of 4 - 6) assume a constant variance.

In [63]:
cyl_4 = mtcars.query('cyl == 4')[['mpg']]
cyl_6 = mtcars.query('cyl == 6')[['mpg']]

sm.CompareMeans(sm.DescrStatsW(cyl_4), sm.DescrStatsW(cyl_6)).tconfint_diff()

(array([ 3.1542863]), array([ 10.68727214]))

**Question** - If someone put a gun to your head and said “Your confidence interval must contain what it’s
estimating or I’ll pull the trigger”, what would be the smart thing to do?
1. Make your interval as wide as possible.
2. Make your interval as small as possible.
3. Call the authorities. 

1. This is just an example of what happens to confidence intervals as you increase the confidence level. You want to be quite sure in your interval (i.e. have a large confidence level) and so you would increase the interval's width

**Question** - Refer back to comparing MPG for 4 versus 6 cylinders. What do you conclude?
1. The interval is above zero, suggesting 6 is better than 4 in the terms of MPG.
2. The interval is above zero, suggesting 4 is better than 6 in the terms of MPG.
3. The interval does not tell you anything about the hypothesis test; you have to do the
test.
4. The interval contains 0 suggesting no difference.



2 The interval was conducted subtracting 4 - 6 and was entirely above zero.

**Question** - Suppose that 18 obese subjects were randomized, 9 each, to a new diet pill and a placebo.
Subjects’ body mass indices (BMIs) were measured at a baseline and again after having
received the treatment or placebo for four weeks. The average difference from follow-up to
the baseline (follow-up - baseline) was 3 kg/m^2 for the treated group and 1 kg/m^2 for the
placebo group. The corresponding standard deviations of the differences was 1.5 kg/m^2 for
the treatment group and 1.8 kg/m^2 for the placebo group. The study aims to answer whether
the change in BMI over the four week period appear to differ between the treated and placebo
groups. What is the pooled variance estimate? 

In [69]:
n1 = 9
n2 = 9
x1 = -3  ##treated
x2 = 1  ##placebo
s1 = 1.5  ##treated
s2 = 1.8  ##placebo

((n1 - 1) * s1**2 + (n2 - 1) * s2**2) / (n1 + n2 -2)

2.745

# Hypothesis Testing

**Question** - Which hypothesis is typically assumed to be true in hypothesis testing?
1. The null.
2. The alternative.

1

**Question** - The type I error rate controls what?

A type I error is the incorrect rejection of a true null hypothesis (a "false positive")

**Question** - Load the data set `mtcars` in the ggplot module. Assume that the data set mtcars is a
random sample. Compute the mean MPG, $\bar{x}$; of this sample. You want to test whether the true
MPG is $\mu_0 $ or smaller using a one sided 5% level test. (H0 : $\mu$ = $\mu_0 $ versus $H_a $ : $\mu$ < $\mu_0 $).
Using that data set and a Z test: Based on the mean MPG of the sample $\bar{x}$; and by using a Z
test: what is the smallest value of $\mu_0 $ that you would reject for (to two decimal places)?

We want to solve
$$ \frac{\sqrt{n}(\bar{x} - \mu_0 )}{s} = Z_{0.05} $$
Or
$$ \mu_0 = \bar{x} - Z_{0.05}s\sqrt{n} = \mu_0 = \bar{x} + Z_{0.95}s\sqrt{n} $$

Note that the quantile is negative.

In [74]:
import numpy as np
from scipy.stats import norm
from ggplot import mtcars as mtcars

mn = np.mean(mtcars[['mpg']])
s = np.std(mtcars[['mpg']])
z = norm().ppf(0.05)

mn - z * s / np.sqrt(len(mtcars))

mpg    21.815492
dtype: float64

Note, it's easy to get tripped up in this problem on signs. If you get a value that's less than $\bar{x}$, then clearly it's wrong, since you'd never reject for $H_0:\mu = \mu_0$ versus $H_a:\mu<\mu_0$ if $\mu_0$ was less than your observed mean. Also note the answer to "What is the largest value for which you would reject for?" is infinity.

**Question** - Consider again the `mtcars` dataset. Use a two group t-test to test the hypothesis that the 4 and
6 cyl cars have the same mpg. Use a two sided test with unequal variances. Do you reject?

In [77]:
import pandas as pd
import statsmodels.stats.api as sm
cyl_4 = mtcars.query('cyl == 4')[['mpg']]
cyl_6 = mtcars.query('cyl == 6')[['mpg']]

sm.ttest_ind(cyl_4, cyl_6, usevar = 'unequal')

(array([ 4.7190594]), array([ 0.00040485]), array([ 12.95598138]))

The above indicates a p-value of 0.0004 indicating to reject at the 5% level.

**Question** - A sample of 100 men yielded an average [PSA](https://www.cancer.gov/types/prostate/psa-fact-sheet) level of 3.0 with a sd of 1.1. What are the complete set of values that a 5% two sided Z test of $H_0: \mu = \mu_0$ would fail to reject the null hypothesis for? 

In [91]:
myNorm = norm(loc = 3, scale = 1.1/10)

myNorm.ppf(0.95)

3.1809338989646618

**Question** - You believe the coin that you’re flipping is biased towards heads. You get 55 heads out of 100
flips. Do you reject at the 5% level that the coin is fair?

In [12]:
from scipy.stats import binom
myBinom = binom(n = 100, p = 0.5)
1 - myBinom.cdf(54)

0.18410080866334821

No you do not reject.

**Question** - Suppose that in an AB test, one advertising scheme led to an average of 10 purchases per day
for a sample of 100 days, while the other led to 11 purchases per day, also for a sample of
100 days. Assuming a common standard deviation of 4 purchases per day. Assuming that the
groups are independent and that they days are iid, perform a Z test of equivalence. Do you
reject at the 5% level? 

In [15]:
from scipy.stats import norm

m1 = 10
m2 = 11
n1 = 100
n2 = 100
s = 4
se = s * np.sqrt(1 / n1 + 1 / n2)
ts = (m2 - m1) / se
2 * norm.cdf(-np.abs(ts))

0.077099871743541701

No you do not reject.

**Question** - A confidence interval for the mean contains:
1. All of the values of the hypothesized mean for which we would fail to reject with alpha = 1 - Confidence Interval.

2. All of the values of the hypothesized mean for which we would fail to reject with 2 * alpha = 1 - Confidence Interval.

3. All of the values of the hypothesized mean for which we would reject with alpha = 1 - Confidence Interval.

4. All of the values of the hypothesized mean for which we would reject with 2 * alpha = 1 - Confidence Interval.


1

The only complicated part of this is the 2. Note that a 95% interval corresponds to a 5% level two sided test. So it's $\alpha = 1 - Conf Level$. The confusion is that for both the two sided test and confidence interval, one needs to calculate $Z_1 - \alpha / 2$ (or the relevant T quantile).

**Question** - In a court of law, all things being equal, if via policy you require a lower standard of evidence
to convict people then
1 Less guilty people will be convicted.
2. More innocent people will be convicted.
3. More Innocent people will be not convicted.

1
If you require less evidence to convict, then you will convict more people, guilty and innocent. Relate this property back to hypothesis tests.

# P-values

**Question** - P-values are probabilities that are calculated assuming which hypothesis is true?
1. the alternative
2. the null

2

**Question** - You get a P-value of 0.06. Would you reject for a type I error rate of 0.05?
1. Yes you would reject the null
2. No you would not reject the null
3. It depends on information not given

2

**Question** - The proposed procedure for getting a two sided P-value for the exact binomial test considered
here is what?
1. Multiplying the one sided P-value by one half
2. Doubling the larger of the two one sided P-values
3. Doubling the smaller of the two one sided P-values
4. No procedure exists

2, but note that this is only possible for the case that p = 0.5

**Question** - Consider again the `mtcars` dataset. Use a two group t-test to test the hypothesis that the 4 and
6 cyl cars have the same mpg. Use a two sided test with unequal variances. Give a P-value.

In [19]:
import statsmodels.stats.api as sm
from ggplot import mtcars as mtcars

cyl_4 = mtcars.query('cyl == 4')[['mpg']]
cyl_6 = mtcars.query('cyl == 6')[['mpg']]

sm.CompareMeans(sm.DescrStatsW(cyl_4), sm.DescrStatsW(cyl_6)).ttest_ind(usevar = 'unequal')[1]

array([ 0.00040485])

Since the p-value is less than 0.05, the null hypothesis can be rejected.

**Question** - You believe the coin that you’re flipping is biased towards heads. You get 55 heads out of 100
flips. Give an exact P-value for the hypothesis that the coin is fair.

In [21]:
from scipy.stats import binom
myBinom = binom(n = 100, p = 0.5)
1 - myBinom.cdf(54)

0.18410080866334821

p-value = 0.1841, thus the null hypothesis cannot be rejected.

**Question** - A web site was monitored for a year and it received 520 hits per day. In the first 30 days in the
next year, the site received 15,800 hits. Assuming that web hits are Poisson. Give an exact one
sided P-value to the hypothesis that web hits are up this year over last. Do you reject? 

In [31]:
from scipy.stats import poisson
myPoisson = poisson(mu = 520 * 30)
1 - myPoisson.cdf(15800)

0.054445702451260791

Does not reject

In [32]:
#Also, compare with the Gaussian approximation where λˆ∼N(λ,λ/t)
import numpy as np
from scipy.stats import norm
myNormal = norm(loc = 520, scale = np.sqrt((520 / 30)))
1 - myNormal.cdf(15800/30)

0.054657288104334234

As t→∞ this becomes more Gaussian. The approximation is pretty good for this setting.

**Question** - Suppose that in an AB test, one advertising scheme led to an average of 10 purchases per day
for a sample of 100 days, while the other led to 11 purchases per day, also for a sample of
100 days. Assuming a common standard deviation of 4 purchases per day. Assuming that the
groups are independent and that they days are iid, perform a Z test of equivalence. Give a
P-value for the test? 

In [36]:
m1 = 10
m2 = 11
n1 = 100
n2 = 100
s = 4
se = s * np.sqrt(1 / n1 + 1 / n2)
ts = (m2 - m1) / se
2 * norm.cdf(-np.abs(ts))

0.077099871743541701

It cannot be rejected.

**Question** - Consider the `mtcars` data set.
1. Give the p-value for a t-test comparing MPG for 6 and 8 cylinder cars assuming equal
variance, as a proportion to 3 decimal places.
2. Give the associated P-value for a z test.
3. Give the common standard deviation estimate for MPG across cylinders to 3 decimal
places.
4. Would the t test reject at the two sided 0.05 level (0 for no 1 for yes)?

In [50]:
# 1
cyl_8 = mtcars.query('cyl == 8')[['mpg']]
cyl_6 = mtcars.query('cyl == 6')[['mpg']]

sm.CompareMeans(sm.DescrStatsW(cyl_8), sm.DescrStatsW(cyl_6)).ttest_ind()[1]

array([ 0.0002947])

So reject at %5 level.

In [51]:
# 2
sm.CompareMeans(sm.DescrStatsW(cyl_8), sm.DescrStatsW(cyl_6)).ztest_ind()[1]

array([  9.91545926e-06])

In [54]:
# 3
n8 = len(cyl_8)
n6 = len(cyl_6)
s8 = np.std(cyl_8)
s6 = np.std(cyl_6)

mixprob = (n8 - 1) / (n8 + n6 - 2)
np.sqrt(mixprob * s8 ** 2  +  (1 - mixprob) * s6 ** 2)

mpg    2.176192
dtype: float64

# Power

** Question ** - Power is a probability calculation assuming which is true:
1. The null hypothesis
2. The alternative hypothesis
3. Both the null and alternative

2, Power is the probability of rejecting the null hypothesis when it is false.

** Question ** - As your sample size gets bigger, all else equal, what do you think would happen to power?

1. It would get larger
2. It would get smaller
3. It would stay the same
4. It cannot be determined from the information given

1.

** Question ** - What happens to power as $\mu_a$ gets further from $\mu_0$?
1. Power decreases
2. Power increases
3. Power stays the same
4. Power oscillates

2

** Question ** - In the context of calculating power, the effect size is?
1. The null mean divided by the standard deviation
2. The alternative mean divided by the standard error
3. The difference between the null and alternative means divided by the standard deviation
4. The standard error divided by the null mean

3

** Question ** - Recall this problem “Suppose that in an AB test, one advertising scheme led to an average of 10 purchases per day for a sample of 100 days, while the other led to 11 purchases per day, also for a sample of 100 days. Assuming a common standard deviation of 4 purchases per day.” Assuming that 10 purchases per day is a benchmark null value, that days are iid and that the standard deviation is 4 purchases for day. Suppose that you plan on sampling 100 days. What would be the power for a one sided 5% Z mean test that purchases per day have increased under the alternative of $\mu = 11$ purchase per day?

In [4]:
from scipy.stats import norm
1 - norm(loc = 11, scale = 0.4).cdf(10 + norm().ppf(0.95) * 0.4)

0.80376494001549381

** Question ** - Researchers would like to conduct a study of healthy adults to detect a four year mean brain volume loss of .01 mm3. Assume that the standard deviation of four year volume loss in this population is .04 $mm^3$. What is necessary sample size for the study for a 5% one sided test versus a null hypothesis of no volume loss to achieve 80% power?

Looking at the hint we set

$$\frac{0.1}{s/\sqrt{n}} + z_{.95} = z_{.2}$$

$$\frac{{({z_{.95} - z_{.2}})^2}s^2}{.01^2} = \frac{{({z_{.95} + z_{.2}})^2}s^2}{.01^2}$$

In [8]:
(norm().ppf(0.95) + norm().ppf(0.8))** 2 * 0.04** 2 / 0.01**2

98.920915712316258

# The bootstrap and resampling

** Question ** - The bootstrap uses what to estimate the sampling distribution of a statistic?
1. The true population distribution
2. The empirical distribution that puts probability 1/n for each observed data point


2

** Question ** - When performing the bootstrap via Monte Carlo resampling for a data set of size n which is true? Assume that you’re going to do 10,000 bootstrap resamples?
1. You sample n complete data sets of size 10,000 with replacement
2. You sample 10,000 complete data sets of size n without replacement
3. You sample 10,000 complete data sets of size n with replacement
4. You sample n complete data sets of size 10,000 without replacement

3

** Question ** - Permutation tests do what?
1. Creates a null distribution for a hypothesis test by permuting a predictor variable.
2. Creates a null distribution by resampling from the response with replacement.
3. Creates an alternative distribution by permuting group labels.
4. Creates confidence intervals by resampling with replacement.

1